downloader/downloader.go

```diff
@@ -40,59 +39,120 @@ var stats struct {
 	downloaded int
 }
 
-func getFileNameByTime(item *photoslibrary.MediaItem) (string, error) {
+//LibraryItem Google Photo item and meta data
+type LibraryItem struct {
+	//Google Photos item
+	Item *photoslibrary.MediaItem
+
+	//Actual file name that was used, without a path
+	UsedFileName string
+}
+
+//MarshalJSON Convert LibraryItem to JSON utlizing pre-built Google marshalling
+func (s *LibraryItem) MarshalJSON() ([]byte, error) {
+	type NoMethod LibraryItem
+	raw := NoMethod(*s)
+	return gensupport.MarshalJSON(raw, []string{}, []string{})
+}
+
+// getFolderPath Path of the to store JSON and image files for the particular MediaItem
+func getFolderPath(item *photoslibrary.MediaItem) string {
+	backupFolder := Options.BackupFolder
```

Comment:
better to put that in an `init` function

Reply:
or just use `Options.BackupFolder` when needed... ?  :thinking:

Reply:
`filepath.Join` gets confused when `backupFolder` is `""`